eng/pipelines/templates/stages/archetype-java-release-batch.yml

```diff
@@ -0,0 +1,508 @@
+
+                  - template: /eng/common/pipelines/templates/steps/create-pull-request.yml
+                    parameters:
+                      PRBranchName: increment-package-version-${{ parameters.ServiceDirectory }}-$(Build.BuildId)
+                      CommitMsg: "Increment package versions for ${{ parameters.ServiceDirectory }} releases"
+                      PRTitle: "Increment versions for ${{ parameters.ServiceDirectory }} releases"
+                      PRLabels: "auto-merge"
+                      CloseAfterOpenForTesting: '${{ parameters.TestPipeline }}'
+
+        - deployment: PublishDocsMs
+          displayName: Docs.MS Release
+          condition: and(succeeded(), ne(variables['Skip.PublishDocs'], 'true'))
+          environment: maven
+          dependsOn: PublishPackage
+
+          pool:
+            name: azsdk-pool-mms-ubuntu-2004-general
+            vmImage: MMSUbuntu20.04
+
+          variables:
+            - template: /eng/pipelines/templates/variables/globals.yml
+            - name: DocValidationImageId
+              value: azuresdkimages.azurecr.io/javarefautocr:latest
+
+          strategy:
+            runOnce:
+              deploy:
+                steps:
+                  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+                    parameters:
+                      SkipDefaultCheckout: true
+                      Paths:
+                        - sdk
+                        - common/perf-test-core
+                        - .github/CODEOWNERS
+                        - '!sdk/**/test-recordings'
+                        - '!sdk/**/session-records'
+                  - download: current
+                    displayName: 'Download Artifact: ${{parameters.ArtifactName}}'
+                    artifact: ${{parameters.ArtifactName}}
+                  - template: /eng/common/pipelines/templates/steps/update-docsms-metadata.yml
+                    parameters:
+                      PackageInfoLocations:
+                        - ${{ each artifact in parameters.Artifacts }}:
+                          - ${{if ne(artifact.skipPublishDocMs, 'true')}}:
+                            - $(Pipeline.Workspace)/${{parameters.ArtifactName}}/PackageInfo/${{artifact.name}}.json
+                      WorkingDirectory: $(System.DefaultWorkingDirectory)
+                      TargetDocRepoOwner: 'Azure'
+                      TargetDocRepoName: ${{parameters.TargetDocRepoName}}
+                      Language: 'java'
+                      SparseCheckoutPaths:
+                        - docs-ref-services/
+                        - metadata/
+                      DocValidationImageId: "$(DocValidationImageId)"
+
+                  - ${{ each artifact in parameters.Artifacts }}:
+                    - ${{if ne(artifact.skipPublishDocMs, 'true')}}:
+                      - template: /eng/pipelines/templates/steps/fetch-package-list.yml
```

Comment:
[in follow-up] We should try to avoid duplicating these steps. This looks to be all in the java repo so should be easier to fix the template.